tests/test_scalars.py

```diff
@@ -156,116 +181,131 @@ def test_timedelta_add_sub() -> None:
     # TypeError: as1, as2, as3, as4, as5, as10, as11, as13
     check(assert_type(td - td, pd.Timedelta), pd.Timedelta)
-    check(assert_type(td - as6, pd.Timedelta), pd.Timedelta)
-    check(assert_type(td - as7, pd.Timedelta), pd.Timedelta)
-    check(assert_type(td - as8, pd.TimedeltaIndex), pd.TimedeltaIndex)
-    check(assert_type(td - as9, pd.Series), pd.Series)
-    check(assert_type(td - as12, npt.NDArray[np.timedelta64]), np.ndarray)
-    check(assert_type(td - as14, NaTType), NaTType)
-    check(assert_type(as1 - td, pd.Period), pd.Period)
-    check(assert_type(as2 - td, pd.Timestamp), pd.Timestamp)
-    check(assert_type(as3 - td, dt.datetime), dt.datetime)
-    check(assert_type(as4 - td, dt.date), dt.date)
-    check(assert_type(as5 - td, pd.Timestamp), pd.Timestamp)
-    # pyright is wrong here because as6 + td calls td.__rsub__(as6),
-    # not timedelta.__sub__
+    check(assert_type(td - as_dt_timedelta, pd.Timedelta), pd.Timedelta)
+    check(assert_type(td - as_timedelta64, pd.Timedelta), pd.Timedelta)
+    check(assert_type(td - as_timedelta_index, pd.TimedeltaIndex), pd.TimedeltaIndex)
+    check(assert_type(td - as_timedelta_series, TimedeltaSeries), pd.Series)
+    check(
+        assert_type(td - as_ndarray_td64, npt.NDArray[np.timedelta64]),
+        np.ndarray,
+        np.timedelta64,
+    )
+    check(assert_type(td - as_nat, NaTType), NaTType)
+    check(assert_type(as_period - td, pd.Period), pd.Period)
+    check(assert_type(as_timestamp - td, pd.Timestamp), pd.Timestamp)
+    check(assert_type(as_datetime - td, dt.datetime), dt.datetime)
+    check(assert_type(as_date - td, dt.date), dt.date)
+    check(assert_type(as_datetime64 - td, pd.Timestamp), pd.Timestamp)
+    # pyright can't know that as6 + td calls td.__rsub__(as6),  not timedelta.__sub__
+    # https://github.com/microsoft/pyright/issues/4088
     check(
         assert_type(
-            as6 - td,  # pyright: ignore[reportGeneralTypeIssues]
+            as_dt_timedelta - td,  # pyright: ignore[reportGeneralTypeIssues]
             pd.Timedelta,
         ),
         pd.Timedelta,
     )
-    check(assert_type(as7 - td, pd.Timedelta), pd.Timedelta)
-    check(assert_type(as8 - td, pd.TimedeltaIndex), pd.TimedeltaIndex)
-    check(assert_type(as9 - td, Union[TimestampSeries, TimedeltaSeries]), pd.Series)
-    check(assert_type(as10 - td, pd.PeriodIndex), pd.PeriodIndex)
-    check(assert_type(as11 - td, pd.DatetimeIndex), pd.DatetimeIndex)
+    check(assert_type(as_timedelta64 - td, pd.Timedelta), pd.Timedelta)
+    check(assert_type(as_timedelta_index - td, pd.TimedeltaIndex), pd.TimedeltaIndex)
+    check(assert_type(as_timedelta_series - td, TimedeltaSeries), pd.Series)
+    check(assert_type(as_period_index - td, pd.PeriodIndex), pd.PeriodIndex)
+    check(assert_type(as_datetime_index - td, pd.DatetimeIndex), pd.DatetimeIndex)
     # pyright is wrong here because ndarray.__sub__(Timedelta) is NotImplemented
```

Comment:
I think it is about as clear as one can get.

```
In [4]: a = np.array([np.timedelta64(1)])
In [5]: import pandas as pd
In [8]: b = pd.Timedelta(1)
In [10]: a.__sub__(b)
Out[10]: NotImplemented
```

`NotImplemented` is what a function returns when it doesn't support a type so that the `__r<func>__` can be tried.

Reply:
Can you check if this has been fixed in pyright 1.1.278?

Reply:
These are fixed, so ignores removed.